src/PID/MotorControllerEnhanced.java

```diff
@@ -0,0 +1,73 @@
+package PID;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.IMotorControllerEnhanced;
+
+public class MotorControllerEnhanced implements PIDController {
+  private IMotorControllerEnhanced motorControllerEnhanced;
+  private PIDFTerms pidfTerms;
+
+  public MotorControllerEnhanced(IMotorControllerEnhanced motorControllerEnhanced) {
+    this.motorControllerEnhanced = motorControllerEnhanced;
+    this.pidfTerms = new PIDFTerms(0, 0, 0, 0);
+  }
+
+  public MotorControllerEnhanced(IMotorControllerEnhanced motorControllerEnhanced, double kP, double kI, double kD,
+                                 double kF) {
+    this.motorControllerEnhanced = motorControllerEnhanced;
+    this.pidfTerms = new PIDFTerms(kP, kI, kD, kF);
+    this.setPID(kP, kI, kD, kF);
+  }
+
+  public MotorControllerEnhanced(IMotorControllerEnhanced motorControllerEnhanced, double kP, double kI, double kD,
+                                 double kF, double setpoint) {
+    this.motorControllerEnhanced = motorControllerEnhanced;
+    PIDFTerms pidfTerms = new PIDFTerms(kP, kI, kD, kF);
+    this.setPID(kP, kI, kD, kF);
+    this.setSetpoint(setpoint);
+  }
+
+
+  @Override
+  public PIDFTerms getPID() {
+    return this.pidfTerms;
+  }
+
+  @Override
+  public double getSetpoint() {
+    return motorControllerEnhanced.getClosedLoopTarget(0);
```

Comment:
Add a field for the PID slot. We might use another one someday 🤷‍♂️ . Of course also make a constructor setting it to 0, so we won't need to bother setting it unless we want to.